packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SubjectGroupViewer/SubjectGroupViewerContainer.js

```diff
@@ -0,0 +1,297 @@
+const DEFAULT_GRID_ROWS = 3
+
+function storeMapper (stores) {
+  const {
+    setOnZoom,
+    setOnPan
+  } = stores.classifierStore.subjectViewer
+  
+  const {
+    active: activeWorkflow
+  } = stores.classifierStore.workflows
+  
+  const workflowConfig = activeWorkflow && activeWorkflow.configuration || {}
+  
+  const cellWidth = workflowConfig.cell_width || DEFAULT_CELL_WIDTH
+  const cellHeight = workflowConfig.cell_height || DEFAULT_CELL_HEIGHT
+  const cellStyle = workflowConfig.cell_style || DEFAULT_CELL_STYLE
+  const gridColumns = workflowConfig.grid_columns || DEFAULT_GRID_COLUMNS
+  const gridRows = workflowConfig.grid_rows || DEFAULT_GRID_ROWS
+
+  return {
+    cellWidth,
+    cellHeight,
+    cellStyle,
+    gridColumns,
+    gridRows,
+    setOnZoom,
+    setOnPan
+  }
+}
+
+class SubjectGroupViewerContainer extends React.Component {
+  constructor () {
+    super()
+    this.dragMove = this.dragMove.bind(this)
+    this.groupViewer = React.createRef()
+    this.scrollContainer = React.createRef()
+
+    this.state = {
+      images: [],
+      panX: 0,
+      panY: 0,
+      zoom: 1,
+    }
+  }
+
+  componentDidMount () {
+    this.onLoad()
+
+    // Listen for pan and zoom actions outside of this component.
+    // i.e. zoom in/out actions from the image controls. 
+    this.props.setOnPan(this.onPanViaExternalControls.bind(this))
+    this.props.setOnZoom(this.onZoomViaExternalControls.bind(this))
+    
+    this.scrollContainer.current?.addEventListener('wheel', this.onWheel.bind(this))
+  }
+  
+  componentWillUmount () {
+
+    this.setOnPan(() => true)
+    this.setOnZoom(() => true)
+    
+    this.scrollContainer.current?.removeEventListener('wheel', this.onWheel.bind(this))
+  }
+
+  fetchImage (url) {
+    const { ImageObject } = this.props
+    return new Promise((resolve, reject) => {
+      const img = new ImageObject()
+      img.onload = () => resolve(img)
+      img.onerror = reject
+      img.src = url
+      return img
+    })
+  }
+
+  dragMove (event, difference) {
+    this.onDrag && this.onDrag(event, difference)
+  }
+
+  async preload () {
+    const { subject } = this.props
+    if (subject && subject.locations) {
+      // TODO: Validate for allowed image media mime types
+      
+      const imageUrls = subject.locations.map(obj => Object.values(obj)[0])
+      const images = await Promise.all(
+        imageUrls.map(url => this.fetchImage(url))
+      )
+      
+      this.setState({ images })
+    }
+    return
+  }
+
+  getGridSize () {
+    const svg = this.groupViewer.current || {}
+    const { width: clientWidth, height: clientHeight } = svg.getBoundingClientRect && svg.getBoundingClientRect() || {}
+    const { gridRows, gridColumns, cellWidth, cellHeight } = this.props
+    
+    const gridWidth = gridColumns * cellWidth
+    const gridHeight = gridRows * cellHeight
+    
+    return {
+      clientHeight,
+      clientWidth,
+      gridHeight,
+      gridWidth,
```

Comment:
I'm still a bit confused by this code. How does grid width and height differ from the client width and height in the browser window? Looking at the storybook, grid dimensions are set from props, which I think are passed down from the workflow config. I'm missing how it ties in to the original natural height and width of each of the subject images.

In the image viewer code, we want to return the rendered size of the image _and_ its natural size so that the classification records how much the image was scaled by for this volunteer. Here, I was thinking we'd want to record how much each subject image was scaled down in the grid viewer.